### PR TITLE
[FIX] website{,_forum}: wrong forum post link when created from the website

### PR DESCRIPTION
**Pull request by hubvd**
Steps to reproduce
==================

- Website > Go to website
- Forum > New post
- Fill the form and post your question
- Go to the backend
- Website > Forum > Post
- Click on the one you just created
- Click on the link in the chatter
-> It goes to an ir.ui.view model

Cause of the issue
==================

Since the form is posted from the frontend, the inherit_branding is
automatically added to the context. This causes the
data-oe-{model,view,id} to be added to the button.

https://github.com/odoo/odoo/blob/46ce4345c96a58845bfa13dd1a86be391cf0a923/addons/website/models/ir_ui_view.py#L425-L426

Solution
========

- Set a context key `inherit_branding` to False inside
`message_post_with_view`.

- In `_render`, only set the key to True if it is not set.

- Also do the same for `inherit_branding_auto`

opw-3070490

### CODE REVIEW

**Comment by hubvd** [comment]
@rdeodoo What do you think ?

**Review by rdeodoo — COMMENTED** [review]
@odoo/rd-sm-learn Hello, this should be for your interest?
@hubvd Could you please line return at [72 char max](https://github.com/odoo/odoo/pull/83508#discussion_r796442104) in your commit message (both title & description)?

**Comment by tde-banana-odoo** [comment]
Wait, what ? Don't we have any better option to skip that kind of behavior ?

**Comment by hubvd** [comment]
> Wait, what ? Don't we have any better option to skip that kind of behavior ?

I don't think so :/ @rdeodoo Any ideas ?

**Review by rdeodoo — CHANGES_REQUESTED** [review]
Surely this be done in a generic fix at a lowel level, right?
This is not a forum only issue, what about making this "skip branding" behavior somewhere around `message_post_with_view`? There is no case that would require the branding in a mail AFAIK.

Once moved, can't you use the existing context key `inherit_branding` and set it to False? (Maybe with some adaption of `website`'s `_render()` override as you did but check if strictly False to bypass).

Also, would need a test with this fix :+1:

**Comment by hubvd** [comment]
Not sure if this is the right place for the test, but it reproduces the original issue :shrug:

File: addons/mail/models/mail_thread.py [hunk 1]
```diff
@@ -1940,7 +1940,7 @@ def _message_compose_with_view(self, views_or_xmlid, message_log=False, **kwargs
 
     def message_post_with_view(self, views_or_xmlid, **kwargs):
         """ Helper method to send a mail / post a message using a view_id """
-        self._message_compose_with_view(views_or_xmlid, **kwargs)
+        self.with_context(inherit_branding=False)._message_compose_with_view(views_or_xmlid, **kwargs)
```
Comment (tde-banana-odoo) [inline addons/mail/models/mail_thread.py:1943]: This should move into website as an override, as it has no usage in base mail module.

File: addons/website/models/ir_ui_view.py [hunk 2]
```diff
@@ -423,7 +423,8 @@ def _render(self, values=None, engine='ir.qweb', minimal_qcontext=False):
             # in edit mode ir.ui.view will tag nodes
             if not translatable and not self.env.context.get('rendering_bundle'):
                 if editable:
-                    new_context = dict(self._context, inherit_branding=True)
+                    new_context = dict(self._context)
+                    new_context.setdefault("inherit_branding", True)
```
Comment (tde-banana-odoo) [inline addons/website/models/ir_ui_view.py:427]: Shouldn't we do the same in the elif just below ?

File: addons/mail/models/mail_thread.py (hunk 1 above)
Comment (hubvd) [reply addons/mail/models/mail_thread.py]: Done :+1:

File: addons/website/models/__init__.py [hunk 3]
```diff
@@ -29,3 +29,4 @@
 from . import website_form
 from . import website_snippet_filter
 from . import website_visitor
+from . import mail_thread
```
Comment (tde-banana-odoo) [inline addons/website/models/__init__.py:32]: Nitpicking: try to keep alphabetical ordering when possible :)

File: addons/website_forum/tests/test_forum.py [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
-
 from .common import KARMA, TestForumCommon
+from odoo import http
+from odoo.addons.http_routing.models.ir_http import slug
 from odoo.exceptions import UserError, AccessError
+from odoo.tests import HttpCase
 from odoo.tools import mute_logger
 from psycopg2 import IntegrityError
 
-
```
Comment (tde-banana-odoo) [inline addons/website_forum/tests/test_forum.py:9]: You may keep this line (unnecessary diff anyway)

File: addons/website_forum/tests/test_forum.py [hunk 5]
```diff
@@ -1,12 +1,13 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
-
```
Comment (tde-banana-odoo) [inline addons/website_forum/tests/test_forum.py:3]: You may keep this line (unnecessary diff anyway)

File: addons/website_forum/tests/test_forum.py (hunk 4 above)
Comment (hubvd) [reply addons/website_forum/tests/test_forum.py]: Oops my bad, I didn't see that

**Comment by kitan191** [comment]
@robodoo delegate+

**Comment by hubvd** [comment]
@robodoo r+
